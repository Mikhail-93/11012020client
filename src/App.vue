<template>
    <v-app id="inspire">
        <v-navigation-drawer v-model="drawer" app>
            <v-list dense>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-home</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Home</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
                <v-list-item link>
                    <v-list-item-action>
                        <v-icon>mdi-contact-mail</v-icon>
                    </v-list-item-action>
                    <v-list-item-content>
                        <v-list-item-title>Contact</v-list-item-title>
                    </v-list-item-content>
                </v-list-item>
            </v-list>
        </v-navigation-drawer>

        <v-app-bar
                app
                color="indigo"
                dark
        >
            <v-app-bar-nav-icon @click.stop="drawer = !drawer"/>
            <v-toolbar-title>Application</v-toolbar-title>
        </v-app-bar>

        <v-content>
            <v-container class="fill-height pa-0" fluid>
                <LMap ref="map" style="z-index: 0">
                    <l-tile-layer
                            url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
                            attribution='Map data: &copy; <a href="http://www.openstreetmap.org/copyright">OpenStreetMap</a>, <a href="http://viewfinderpanoramas.org">SRTM</a> | Map style: &copy; <a href="https://opentopomap.org">OpenTopoMap</a> (<a href="https://creativecommons.org/licenses/by-sa/3.0/">CC-BY-SA</a>)'
                    >
                    </l-tile-layer>
                    <!--                    <l-control-attribution position="bottomright" prefix="leaflet OpenStreetMap Contributions" >-->

                </LMap>
                <!--                class="align-stretch"-->
            </v-container>
        </v-content>
    </v-app>
</template>

<script>
    import {LMap, LTileLayer, LMarker, LControlAttribution} from 'vue2-leaflet';

    export default {
        components: {
            LMap,
            LTileLayer,
            LMarker,
            LControlAttribution
        },
        props: {
            source: String,
        },
        data: () => ({
            drawer: null,
        }),
        mounted() {
            this.$nextTick(() => {
                // this.map=
                this.$refs.map.mapObject.setView([51.505, -0.09], 13)

            })
        }
    }
</script>