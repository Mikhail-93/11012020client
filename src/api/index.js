import 'isomorphic-fetch'
import fetchIntercept from "fetch-intercept/lib/node";

import errorInterceptor from "./errorInterceptor";
import dataInterceptor from "./dataInterceptor";
import cookieInterceptor from "./cookieInterceptor";
import headersInterceptor from "./headersInterceptor";
import api from "./api";

fetchIntercept.register(errorInterceptor)
fetchIntercept.register(dataInterceptor)
fetchIntercept.register(cookieInterceptor)
fetchIntercept.register(headersInterceptor)



export default api
