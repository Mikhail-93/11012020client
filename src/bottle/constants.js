/**
 * Created by alexey2baranov on 8/20/16.
 *
 * домены и редикт урлы
 * https://vk.com/apps?act=manage
 * дебаг на телефоне
 * https://developers.google.com/web/tools/chrome-devtools/remote-debugging?utm_source=dcc&utm_medium=redirect&utm_campaign=2016q3
 */
let constants = {
  "development": {
    sw: {
      delay: 15000,
    },
    messenger: {
      // идентификатор нашего приложения ВК
      // clientId: 7210289,
      // страница, на которую нужно перейти для аутентификации
      loginUrl: 'http://localhost:8081/connect/vkontakte',
      // с мобилки
      // loginUrl: 'http://192.168.43.84:8081/connect/vkontakte'
    },
    api: {
      path: "http://localhost:8081/api"
      // с мобилки
      // path: "http://192.168.43.84:8081/api"
    },
    di: {
      fetch: true,
      cookie: false
    }
  },
  "test": {
    sw: {
      delay: Number.MAX_SAFE_INTEGER,
    },
    messenger: {
      // clientId: 7210289,
      loginUrl: 'http://localhost:8082/connect/vkontakte',
      // redirectUrl: 'http://localhost:8082/login/check-vk'
    },
    api: {
      path: "http://localhost:8082/api"
    },
    di: {
      fetch: false,
      cookie: true
    }
  },
  "staging": {
    sw: {
      delay: 20000,
    },
    messenger: {
      loginUrl: 'https://staging.kopnik.org/connect/vkontakte',
      // clientId: 7210289,
      // redirectUrl: 'https://staging.kopnik.org/login/check-vk'
    },
    api: {
      path: "https://staging.kopnik.org/api"
    },
    di: {
      fetch: true,
      cookie: false
    }
  }
}

export default constants
