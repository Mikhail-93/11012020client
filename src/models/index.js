/**
 * Created by alexey2baranov on 8/20/16.
 */

import AbstractSync from "./AbstractSync" //для того чтобы правильно работали рекурсивные ссылки модулей
// import Registration from "./model/Registration"
import Kopnik from "./Kopnik"
// import Kopa from "./model/Kopa"
// import File from "./model/File"

export {AbstractSync, Kopnik}