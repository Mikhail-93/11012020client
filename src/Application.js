import {AbstractSync, Kopnik} from "./models";

export default class Application {
    static getInstance() {
        if (!this.instance) {
            this.instance = new this()
        }
        return this.instance
    }

    user = null

    constructor() {
        /**
         * Кэш моделей
         * @type {Array}
         */
        // пока не понятно зачем приложению кэш.
        // он вероятно будет сильно торможить приложение из-за реактивности App.$data в которой он находится
        // хотя он же не в App.$data а в App.$options !!
        // уже в дата потому тчо в options не реактивен
        //this.models = AbstractSync.cache

        /**
         * Идентификатор раздела
         * Равен названиею соответствующего компонента
         *
         * @type {string} Map | Profile | Thanks
         */
        this.SECTION = "Map"

        this.user = undefined

        this.initUser()
    }

    getSharedState() {
        return {
            SECTION: this.SECTION
        }
    }

    setState(state) {
        if (state.SECTION) {
            this.SECTION = state.SECTION
        }
    }

    /**
     * Инициализирует ползователя при запуске приложения
     *
     * @returns {Promise<void>}
     */
    async initUser() {
        let vkUser = localStorage.getItem("vkUser")
        if (vkUser) {
            vkUser = JSON.parse(vkUser)
            let user //= await Kopnik.getVkUserStatus(vkUser.uid)
            if (vkUserStatus) {
                this.user= Kopnik.merge(vkUserStatus.user)
            } else {
                this.setVkUser(vkUser)
                this.SECTION="Profile"
            }
        }
    }

    /**
     * Учстанавливает пользователем объект из VK.init::success()
     * @param vkUser
     * @returns
     */
    setVkUser(vkUser){
        this.user= new Kopnik
        this.user.merge({
            hash: vkUser.hash,
            uid: vkUser.uid,
            firstname: vkUser.first_name,
            surname: vkUser.last_name,
            photo: vkUser.photo,
            smallPhoto: vkUser.photo_rec,
        })
        localStorage.setItem("vkUser", JSON.stringify(vkUser))
    }
}
