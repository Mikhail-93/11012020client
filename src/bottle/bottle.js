import Bottle from "bottlejs"
import _ from "lodash"
// import 'json-form-data'
import * as loglevel from "loglevel";

import Application from "../application/Application"
import constants from './constants'
import api from "../api";
import CookieService from "./CookieService"
import MK from "../mk/mk";
import LocaleManager from "../locales/LocaleManager";
import ru from '../locales/ru'
import en from '../locales/en'

Bottle.config.strict = true
const bottle = new Bottle()

bottle.factory('logger', function loggerFactory() {
    //все плагины ломают стектрейс консоли. то есть невозможно увидеть из какого файла и какой строки был вызван лог!
    // LoglevelPluginPrefix.reg(loglevel)
    // LoglevelPluginPrefix.apply(loglevel, {template: "%t [%l] %n: "})
    loglevel.setLevel(loglevel.levels.DEBUG)
    if (process.env.NODE_ENV == "test") {
        loglevel.setLevel(loglevel.levels.DEBUG)
        // LoglevelPluginToString.apply(loglevel, {})
    }
// Be sure to call setLevel method in order to apply plugin
// loglevel.getLogger("StateManager").setLevel("info")
    return loglevel
})

bottle.constant('messages', {
    ru,
    en,
})

bottle.factory('env', function env() {
    return process.env.NODE_ENV || 'development'
})
bottle.service('localeManager', function localeManager() {
    const localeManager = new LocaleManager()
    localeManager.currentLocale = 'ru'//container.env === 'development' ? 'en' : 'ru'
    return localeManager
})
bottle.service('cookieService', CookieService, 'constants')
bottle.factory('api', function apiFactory(container) {
    if (!container.constants.di.fetch) {
        throw new Error('not supported')
    }
    return api
})
bottle.factory('VK', function vkFactory(container) {
    return process.env.NODE_ENV === 'test' ? MK : global.VK
})
bottle.factory('constants', function constantsFactory() {
    if (!process.env.NODE_ENV) {
        throw new Error("NODE_ENV is not defined")
    }

    let local = {}//require("./local.js")
    let result = _.merge({}, constants, local)[process.env.NODE_ENV]

    return result
})
if (!Application) {
    throw new Error("Application is undefined in bottle")
}
bottle.service('application', Application, 'logger')

/**
 * @callback fetch
 * @param {string} url
 * @param {Object?} options
 */
/**
 * @type {Object}
 *
 * @property {loglevel} logger
 * @property {Application} application
 * @property {Location} Location
 * @property {fetch} api
 * @property {MK.class} VK
 * @property {CookieService} cookieService
 * @property {Object} constants
 * @property {Object} constants.api
 * @property {String} constants.api.path
 * @property {Object} constants.di
 * @property {Boolean} constants.di.fetch
 * @property {Number} constants.di.cookie
 * @property {Number} constants.messenger.clientId
 * @property {String} constants.messenger.loginUrl
 * @property {String} constants.messenger.redirectUrl
 * @property {String} constants.sw
 * @property {String} constants.sw.delay
 * @property {LocaleManager} localeManager
 * @property {string} env
 * @property {{ru, en}} messages
 */
const container = bottle.container
export {bottle, container}
