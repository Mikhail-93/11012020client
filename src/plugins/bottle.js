import Bottle from "bottlejs"
import _ from "lodash"
import 'json-form-data'
import * as loglevel from "loglevel";

import Application from "../Application"
import config from '../../config'
import fetchApi from "../bottle/fetchApi";
import fetchApiMock from "../bottle/fetchApi.mock";

Bottle.config.strict = true
const bottle = new Bottle()

/**
 * Настраивает контейнер
 * Для тестов необходим сброс и перенастройка чтобы соблюсти изолированность тестов
 *
 * @param {Object} options параметры по умолчанию соответствуют наиболее приемлемым для окружения значениям окружению
 * @param {Number=} options.cookie
 * @param {Boolean=} options.fetch
 */
Bottle.prototype.setup = function (options) {
    bottle.factory('defaultFetchApiOptions', function defaultFetchApiOptionsFactory(container) {
        const cookies = {
            1: 'PHPSESSID=user-1',
            2: 'PHPSESSID=eqlae6a1grpvcirpemfvebf806',
            3: 'PHPSESSID=user-3',
            4: 'PHPSESSID=user-4',
            5: 'PHPSESSID=user-5',
            6: 'PHPSESSID=user-6',
            7: 'PHPSESSID=user-7',
            8: 'PHPSESSID=user-8',
            9: 'PHPSESSID=user-9',
            10: 'PHPSESSID=user-10',
        }
        const result = {}
        if (container.config.di.cookie) {
            result.headers = {
                cookie: cookies[container.config.di.cookie]
            }
        }
        // console.log(result)
        return result
    })
    bottle.factory('fetchApi', function fetchApiFactory(container) {
        return container.config.di.fetch ? fetchApi : fetchApiMock
    })
    bottle.factory('config', function configFactory() {
        if (!process.env.NODE_ENV) {
            throw new Error("NODE_ENV is not defined");
        }

        let local = {}//require("./local.js")
        let result = _.merge({}, config, local)[process.env.NODE_ENV]

        return result
    })
    if (!Application){
        throw new Error("Application is undefined in bottle")
    }
    bottle.service('application', Application, 'logger')
    bottle.factory('logger', function loggerFactory() {
        //все плагины ломают стектрейс консоли. то есть невозможно увидеть из какого файла и какой строки был вызван лог!
        // LoglevelPluginPrefix.reg(loglevel)
        // LoglevelPluginPrefix.apply(loglevel, {template: "%t [%l] %n: "})
        loglevel.setLevel(loglevel.levels.DEBUG)
        if (process.env.NODE_ENV == "test") {
            loglevel.setLevel(loglevel.levels.WARN)
            // LoglevelPluginToString.apply(loglevel, {})
        }
// Be sure to call setLevel method in order to apply plugin
// loglevel.getLogger("StateManager").setLevel("info")
        return loglevel
    })
    Object.assign(this.container.config.di, options)
}

bottle.setup()

/**
 * @callback fetch
 * @param {string} url
 * @param {Object} options
 */
/**
 * @type {Object}
 *
 * @property {loglevel} logger
 * @property {Application} application
 * @property {Location} Location
 * @property {fetch} fetchApi
 * @property {Object} defaultFetchApiOptions
 * @property {Object} defaultFetchApiOptions.headers
 * @property {Object} defaultFetchApiOptions.headers.cookie
 * @property {Object} config
 */
const container = bottle.container
export {bottle, container}
