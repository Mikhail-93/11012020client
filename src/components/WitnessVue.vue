<template>
    <v-container fluid class="fill-height k-witness flex-column align-center">
        <v-card v-for="(eachRequest, index) in application.user.witnessRequests" :key="eachRequest.id"
                elevation="12" class="mb-10" width="100%" max-width="350px">
            <kopnik-view :value="eachRequest" locale fio birth-year passport role location readonly></kopnik-view>
            <v-card-actions>
                <v-btn  color="success" @click="updateRequestStatus_click(eachRequest, Status.CONFIRMED)" class="flex-grow-1"
                        v-promise-btn
                >
                    {{$t('witness.confirm')}}
                </v-btn>
                <v-btn color="error"  @click="updateRequestStatus_click(eachRequest, Status.DECLINED)" class="flex-grow-1"
                       v-promise-btn
                >
                    {{$t('witness.decline')}}
                </v-btn>
            </v-card-actions>
        </v-card>
        <div v-if="application.user.witnessRequests && !application.user.witnessRequests.length" class="overline font-weight-bold my-auto" style="font-size: 1.625rem !important; color: #BDBD">Заявок нет</div>
    </v-container>
</template>
<script>
    import KopnikView from "./KopnikVue"
    import {Kopnik} from "../models"
    import {container} from "../bottle/bottle";
    import logger from "./mixin/logger";

    export default {
        name: "Witness",
        mixins: [logger],
        components: {
            KopnikView
        },
        data: () => {
            return {
                application: container.application,
                Status: Kopnik.Status,
            }
        },
        props: {},
        computed: {},
        watch: {},
        methods: {
            /**
             *
             * @param {Kopnik} request
             * @param {Number} status
             * @returns {Promise<void>}
             */
            async updateRequestStatus_click(request, status) {
                request.status = status
                await this.application.user.updateWitnessRequestStatus(request)
            },
        },
        async created() {
            await container.application.user.reloadWitnessRequests()
        },
        async mounted() {
        }
    }
</script>
