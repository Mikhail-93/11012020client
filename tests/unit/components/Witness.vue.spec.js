import flushPromises from "flush-promises";
import Vue from 'vue'
import { mount } from '@vue/test-utils'

import vuePlugins from "../../test-setup";
import {container} from "../../../src/bottle/bottle";
import {Kopnik} from "../../../src/models";
import WitnessVue from "../../../src/components/WitnessVue";

describe('unit components Witness', () => {
    beforeAll(async () => {
    })

    beforeEach(() => {
    })

    it('draw', async () => {
        await login(1)
        await container.application.authenticate()
        const wrapper = mount(WitnessVue, {
            ...vuePlugins,
            propsData:{
            }
        })
        await flushPromises()
        expect(wrapper.vm.$el).toMatchSnapshot()
    })
})

