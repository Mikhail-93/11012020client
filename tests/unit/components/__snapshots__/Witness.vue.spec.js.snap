// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`unit components Witness draw 1`] = `
<div
  class="container fill-height k-witness flex-column align-center container--fluid"
>
  <div
    class="mb-10 v-card v-sheet theme--light elevation-12"
    style="max-width: 350px; width: 100%;"
  >
    <div
      birth-year=""
      class="v-list v-sheet v-sheet--tile theme--light"
      role="list"
    >
      <div
        class="v-list-item theme--light"
        role="listitem"
        tabindex="-1"
      >
        <span
          class="v-badge {avatarMxAuto: 'mx-auto'} v-badge--bottom theme--light"
        >
          <div
            class="v-avatar"
            style="height: 128px; min-width: 128px; width: 128px;"
          >
            <img
              src="https://lorempixel.com/200/200/?47557"
              style="object-fit: cover;"
            />
          </div>
          <span
            class="v-badge__wrapper"
          >
            <span
              aria-atomic="true"
              aria-label="знак"
              aria-live="polite"
              class="v-badge__badge orange"
              role="status"
            >
              1
            </span>
          </span>
        </span>
      </div>
       
      <!---->
       
      <div
        class="v-list-item theme--light"
        role="listitem"
        tabindex="-1"
      >
        <div
          class="v-list-item__content"
        >
          <div
            class="v-input v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted v-select v-autocomplete"
          >
            <div
              class="v-input__control"
            >
              <div
                aria-expanded="false"
                aria-haspopup="listbox"
                aria-owns="list-14"
                class="v-input__slot"
                role="combobox"
              >
                <div
                  class="v-select__slot"
                >
                  <label
                    class="v-label v-label--active theme--light"
                    for="input-14"
                    style="left: 0px; position: absolute;"
                  >
                    Язык / Language
                  </label>
                  <input
                    autocomplete="off"
                    id="input-14"
                    type="text"
                  />
                  <div
                    class="v-input__append-inner"
                  >
                    <div
                      class="v-input__icon v-input__icon--append"
                    >
                      <i
                        aria-hidden="true"
                        class="v-icon notranslate mdi mdi-menu-down theme--light"
                      />
                    </div>
                  </div>
                  <input
                    type="hidden"
                    value="ru"
                  />
                </div>
                <div
                  class="v-menu"
                >
                  <div
                    class="v-menu__content theme--light v-autocomplete__content"
                    style="max-height: 304px; min-width: 0px; max-width: auto; top: 12px; left: 0px; transform-origin: top left; z-index: 0; display: none;"
                  />
                </div>
              </div>
              <div
                class="v-text-field__details"
              >
                <div
                  class="v-messages theme--light"
                >
                  <span
                    class="v-messages__wrapper"
                    name="message-transition"
                    tag="div"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="v-list-item theme--light"
        role="listitem"
        tabindex="-1"
      >
        <div
          class="v-list-item__content"
        >
          <span>
            <div
              class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted success--text"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light success--text"
                      for="input-22"
                      style="left: 0px; position: absolute;"
                    >
                      Фамилия
                    </label>
                    <input
                      id="input-22"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light success--text"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </span>
        </div>
      </div>
       
      <div
        class="v-list-item theme--light"
        role="listitem"
        tabindex="-1"
      >
        <div
          class="v-list-item__content"
        >
          <span>
            <div
              class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted success--text"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light success--text"
                      for="input-27"
                      style="left: 0px; position: absolute;"
                    >
                      Имя
                    </label>
                    <input
                      id="input-27"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light success--text"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </span>
        </div>
      </div>
       
      <div
        class="v-list-item theme--light"
        role="listitem"
        tabindex="-1"
      >
        <div
          class="v-list-item__content"
        >
          <span>
            <div
              class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted success--text"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light success--text"
                      for="input-32"
                      style="left: 0px; position: absolute;"
                    >
                      Отчество
                    </label>
                    <input
                      id="input-32"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light success--text"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                </div>
              </div>
            </div>
          </span>
        </div>
      </div>
       
      <!---->
       
      <div
        class="v-list-item theme--light"
        role="listitem"
        tabindex="-1"
      >
        <div
          class="v-list-item__content"
        >
          <span>
            <div
              class="v-input v-input--has-state v-input--is-label-active v-input--is-dirty theme--light v-text-field v-text-field--is-booted success--text"
            >
              <div
                class="v-input__control"
              >
                <div
                  class="v-input__slot"
                >
                  <div
                    class="v-text-field__slot"
                  >
                    <label
                      class="v-label v-label--active theme--light success--text"
                      for="input-37"
                      style="left: 0px; position: absolute;"
                    >
                      Последние 4 цифры пасспорта
                    </label>
                    <input
                      id="input-37"
                      type="text"
                    />
                  </div>
                </div>
                <div
                  class="v-text-field__details"
                >
                  <div
                    class="v-messages theme--light success--text"
                  >
                    <span
                      class="v-messages__wrapper"
                      name="message-transition"
                      tag="div"
                    />
                  </div>
                  <div
                    class="v-counter theme--light"
                  >
                    4 / 4
                  </div>
                </div>
              </div>
            </div>
          </span>
        </div>
      </div>
       
      <div
        class="v-list-item theme--light"
        role="listitem"
        tabindex="-1"
      >
        <div
          class="v-list-item__content"
        >
          <div
            class="vue2leaflet-map leaflet-container leaflet-grab leaflet-touch-drag"
            style="z-index: 0; height: 50vh; position: relative;"
            tabindex="0"
          >
            <div />
            <div />
            <div />
             
            <!---->
            <div
              class="leaflet-pane leaflet-map-pane"
              style="left: 0px; top: 0px;"
            >
              <div
                class="leaflet-pane leaflet-tile-pane"
              >
                <div
                  class="leaflet-layer "
                  style="z-index: 1;"
                >
                  <div
                    class="leaflet-tile-container leaflet-zoom-animated"
                    style="z-index: 18; left: 0px; top: 0px;"
                  >
                    <img
                      alt=""
                      class="leaflet-tile"
                      role="presentation"
                      src="https://a.tile.openstreetmap.org/14/14974/9629.png"
                      style="width: 256px; height: 256px; left: -91px; top: -40px;"
                    />
                  </div>
                </div>
              </div>
              <div
                class="leaflet-pane leaflet-shadow-pane"
              >
                <img
                  alt=""
                  class="leaflet-marker-shadow leaflet-zoom-hide"
                  src="marker-shadow.png"
                  style="margin-left: -12px; margin-top: -41px; width: 41px; height: 41px; left: 0px; top: 0px;"
                />
              </div>
              <div
                class="leaflet-pane leaflet-overlay-pane"
              />
              <div
                class="leaflet-pane leaflet-marker-pane"
              >
                <img
                  alt=""
                  class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
                  src="marker-icon.png"
                  style="margin-left: -12px; margin-top: -41px; width: 25px; height: 41px; left: 0px; top: 0px; z-index: 0;"
                  tabindex="0"
                />
              </div>
              <div
                class="leaflet-pane leaflet-tooltip-pane"
              />
              <div
                class="leaflet-pane leaflet-popup-pane"
              />
            </div>
            <div
              class="leaflet-control-container"
            >
              <div
                class="leaflet-top leaflet-left"
              >
                <div
                  class="leaflet-control-zoom leaflet-bar leaflet-control"
                >
                  <a
                    aria-label="Zoom in"
                    class="leaflet-control-zoom-in"
                    href="#"
                    role="button"
                    title="Zoom in"
                  >
                    +
                  </a>
                  <a
                    aria-label="Zoom out"
                    class="leaflet-control-zoom-out"
                    href="#"
                    role="button"
                    title="Zoom out"
                  >
                    −
                  </a>
                </div>
              </div>
              <div
                class="leaflet-top leaflet-right"
              />
              <div
                class="leaflet-bottom leaflet-left"
              />
              <div
                class="leaflet-bottom leaflet-right"
              >
                <div
                  class="leaflet-control-locate leaflet-bar leaflet-control"
                >
                  <a
                    class="leaflet-bar-part leaflet-bar-part-single"
                    title="Определить мое местоположение"
                  >
                    <span
                      class="fa fa-map-marker"
                    />
                  </a>
                </div>
                <div
                  class="leaflet-control-attribution leaflet-control"
                >
                  <a
                    href="https://leafletjs.com"
                    title="A JS library for interactive maps"
                  >
                    Leaflet
                  </a>
                   | Map data: © 
                  <a
                    href="http://www.openstreetmap.org/copyright"
                  >
                    OpenStreetMap
                  </a>
                  , 
                  <a
                    href="http://viewfinderpanoramas.org"
                  >
                    SRTM
                  </a>
                   | Map style: © 
                  <a
                    href="https://opentopomap.org"
                  >
                    OpenTopoMap
                  </a>
                   (
                  <a
                    href="https://creativecommons.org/licenses/by-sa/3.0/"
                  >
                    CC-BY-SA
                  </a>
                  )
                </div>
              </div>
            </div>
             
            <!---->
             
            <div
              style="display: none;"
            />
             
            <!---->
          </div>
        </div>
      </div>
       
    </div>
     
    <div
      class="v-card__actions"
    >
      <button
        class="flex-grow-1 v-btn v-btn--contained theme--light v-size--default"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          
                Заверить
            
        </span>
      </button>
       
      <button
        class="flex-grow-1 v-btn v-btn--contained theme--light v-size--default"
        type="button"
      >
        <span
          class="v-btn__content"
        >
          
                Отклонить
            
        </span>
      </button>
    </div>
  </div>
   
  <!---->
</div>
`;
